src/matroskavideo.cpp

```diff
@@ -0,0 +1,800 @@
+  xmpData_["Xmp.video.MimeType"] = mimeType();
+
+  while (continueTraversing_)
+    decodeBlock();
+
+  aspectRatio();
+}
+
+void MatroskaVideo::decodeBlock() {
+  byte buf[8];
+  io_->read(buf, 1);
+
+  if (io_->eof()) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  uint32_t sz = findBlockSize(buf[0]);  // 0-8
+  if (sz > 0)
+    io_->read(buf + 1, sz - 1);
+
+  const MatroskaTags* mt = findTag(matroskaTags, returnTagValue(buf, sz));
+
+  if (!mt) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  if (mt->first == 0xc53bb6b || mt->first == 0xf43b675) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  bool skip = find(compositeTagsList, mt->first) != 0;
+  bool ignore = find(ignoredTagsList, mt->first) != 0;
+
+  io_->read(buf, 1);
+  sz = findBlockSize(buf[0]);  // 0-8
+
+  if (sz > 0)
+    io_->read(buf + 1, sz - 1);
+  uint64_t size = returnTagValue(buf, sz);
+
+  if (skip && !ignore)
+    return;
+
+  const uint64_t bufMinSize = 200;
+
+#ifndef SUPPRESS_WARNINGS
+  if (!ignore && size > bufMinSize) {
+    EXV_WARNING << "Size " << size << " of Matroska tag 0x" << std::hex << mt->first << std::dec << " is greater than "
+                << bufMinSize << ": ignoring it.\n";
+  }
+#endif
+  if (ignore || size > bufMinSize) {
+    io_->seek(size, BasicIo::cur);
+    return;
+  }
+
+  DataBuf buf2(bufMinSize + 1);
+  std::fill(buf2.begin(), buf2.end(), 0x0);
+  size_t s = static_cast<size_t>(size);
+  io_->read(buf2.data(), s);
+  contentManagement(mt, buf2.data(), s);
+}  // MatroskaVideo::decodeBlock
+
+void MatroskaVideo::contentManagement(const MatroskaTags* mt, const byte* buf, size_t size) {
+  int64_t duration_in_ms = 0;
+  static double time_code_scale = 1.0, temp = 0;
+  static uint64_t stream = 0, track_count = 0;
+  char str[4] = "No";
+  const MatroskaTags* internalMt = 0;
+
+  switch (mt->first) {
+    case 0x0282:
+    case 0x0d80:
+    case 0x1741:
+    case 0x3ba9:
+    case 0x066e:
+    case 0x0660:
+    case 0x065c:
+    case 0x067e:
+    case 0x047a:
+    case 0x0487:
+    case 0x05a3:
+    case 0x136e:
+    case 0x23ca:
+    case 0xeb524:
+      xmpData_[mt->second] = buf;
+      break;
+
+    case 0x0030:
+    case 0x003a:
+    case 0x0287:
+    case 0x14b0:
+    case 0x14ba:
+    case 0x285:
+    case 0x06ae:
+    case 0x0286:
+    case 0x02f7:
+    case 0x2264:
+    case 0x14aa:
+    case 0x14bb:
+    case 0x14cc:
+    case 0x14dd:
+      xmpData_[mt->second] = returnValue(buf, size);
+
+      if (mt->first == 0x0030 || mt->first == 0x14b0) {
+        width_ = returnValue(buf, size);
+      } else if (mt->first == 0x003a || mt->first == 0x14ba) {
+        height_ = returnValue(buf, size);
+      }
+      break;
+
+    case 0x001a:
+    case 0x001f:
+    case 0x0254:
+    case 0x07e1:
+    case 0x07e5:
+    case 0x07e6:
+    case 0x1033:
+    case 0x14b2:
+    case 0x14b3:
+    case 0x23c3:
+    case 0x29bf:
+    case 0x3e8a:
+    case 0x3e9a:
+      switch (mt->first) {
```

Comment:
[suggestion] Since this function is quite large, it would help to create sub-functions with proper names that would help in 2 ways:
* make this function shorter
* Understand which kind of action is going on for each category of tags.

Reply:
Thank you for your note ,
I changed this function  here 2da2a0a8d360decd8afb688f7b751e6395942f66 , It's more readable now and shorter